scripts/workers/code_upload_submission_worker.py

```diff
@@ -127,14 +171,76 @@ def create_job_object(message, environment_image):
     # Create the specification of deployment
     spec = client.V1JobSpec(backoff_limit=1, template=template)
     # Instantiate the job object
-    job = client.V1Job(
-        api_version="batch/v1",
-        kind="Job",
-        metadata=client.V1ObjectMeta(
-            name="submission-{0}".format(submission_pk)
+    job = get_job_object(submission_pk, spec)
+    return job
+
+
+def create_static_code_upload_submission_job_object(message):
+    """Function to create the static code upload pod AWS EKS Job object
+
+    Arguments:
+        message {[dict]} -- Submission message from AWS SQS queue
+
+    Returns:
+        [AWS EKS Job class object] -- AWS EKS Job class object
+    """
+
+    PYTHONUNBUFFERED_ENV = client.V1EnvVar(name="PYTHONUNBUFFERED", value="1")
+    # Used to create submission file by phase_pk and selecting dataset location
+    MESSAGE_BODY_ENV = client.V1EnvVar(name="BODY", value=json.dumps(message))
+    submission_pk = message["submission_pk"]
+    challenge_pk = message["challenge_pk"]
+    phase_pk = message["phase_pk"]
+    image = message["submitted_image_uri"]
+    volume_mount_list = get_volume_mount_list("/dataset")
+    # Get init container
+    init_container = get_init_container(submission_pk)
+    # Configure submission pod container pre-stop hook to submit file.
+    submission_path = "/submission"
+    SUBMISSION_PATH_ENV = client.V1EnvVar(
+        name="SUBMISSION_PATH", value=submission_path
+    )
+    # TODO: Mount an empty folder at submission_path or create post-start hook
+    submission_curl_request = get_static_code_upload_submission_file_curl(
+        challenge_pk, phase_pk, submission_pk, submission_path
+    )
+    pre_stop_exec_command = "mkdir -p {submission_path}; for file_path in `ls {submission_path} | tail -n 1`; do {curl_request}; done;".format(
```

Comment:
@savish28 `submission` folder should be present when initializing the container

Reply:
Yeah, I thought, in case the participant deletes the submission folder(by mistake or something) xD our pre-stop hook method might fail, so added "mkdir -p". I guess if we are checking only submission.json or submission.csv I would refactor it and directly check if the file exists.